sh_script/build_final.sh

```diff
@@ -3,26 +3,39 @@
 export CC=clang
 export AR=llvm-ar
 
-if [[ ! $PWD =~ rust-td$ ]];then
+if [[ ! $PWD =~ td-shim$ ]];then
     pushd ..
 fi
 
-cargo xbuild -p rust-tdshim --target x86_64-unknown-uefi --release
-pushd rust-td-payload
-cargo xbuild --target x86_64-unknown-uefi --release
-popd
-cargo run -p rust-td-tool -- \
-    target/x86_64-unknown-uefi/release/ResetVector.bin \
-    target/x86_64-unknown-uefi/release/rust-tdshim.efi \
-    target/x86_64-unknown-uefi/release/rust-td-payload.efi \
-    target/x86_64-unknown-uefi/release/final.bin
+cargo xbuild -p td-shim --target x86_64-unknown-uefi --release --features=main,tdx --no-default-features
+
+final_pe() {
+    echo final-pe
+    cargo xbuild -p td-payload --target x86_64-unknown-uefi --release --features=main,tdx --no-default-features
+    cargo run -p td-shim-tools --features="linker" --bin td-shim-ld -- \
+        target/x86_64-unknown-uefi/release/ResetVector.bin \
+        target/x86_64-unknown-uefi/release/td-shim.efi \
+        target/x86_64-unknown-uefi/release/td-payload.efi \
+        -o target/x86_64-unknown-uefi/release/final-pe.bin
+}
 
-cargo xbuild -p rust-tdshim --target x86_64-unknown-uefi --release
-pushd rust-td-payload
-cargo xbuild --target target.json --release
-popd
-cargo run -p rust-td-tool -- \
+final_elf() {
+    echo final-elf
+    cargo xbuild -p td-payload --target devtools/rustc-targets/x86_64-unknown-none.json --release --features=main,tdx
+    cargo run -p td-shim-tools --features="linker" --no-default-features --bin td-shim-ld -- \
     target/x86_64-unknown-uefi/release/ResetVector.bin \
-    target/x86_64-unknown-uefi/release/rust-tdshim.efi \
-    target/target/release/rust-td-payload \
-    target/x86_64-unknown-uefi/release/final.bin
+        target/x86_64-unknown-uefi/release/td-shim.efi \
+        target/x86_64-unknown-none/release/td-payload \
+        -o target/x86_64-unknown-uefi/release/final-elf.bin 
+}
+
+if [[ $1 ]];then
```

Comment:
This might be clearer as a `case`, 

```bash
case "${1:-}" in
    elf) final_elf ;;
    pe) final_pe ;;
    *) final_pe && finale_elf ;;
esac
```

Reply:
Done